deploy/internal/deployment-endpoint.yaml

```diff
@@ -23,6 +23,16 @@ spec:
         noobaa.io/configmap-hash: ""
     spec:
       serviceAccountName: noobaa-endpoint
+      topologySpreadConstraints:
+      - maxSkew: 1
+        topologyKey: kubernetes.io/hostname
+        whenUnsatisfiable: ScheduleAnyway
+        # nodeTaintsPolicy available from kubernetes 1.26 or latest, 
```

Comment:
That might be a problem in ODF. 
I see that in 1.26, it is available as beta in Kubernetes, but I don't know the availability of k8s beta features in openshift. @naveenpaul1, any idea?

In any case, as long as this is supported by OCP 4.14, we're good. Otherwise, we can't use it since we should honor the taints policy.
@nimrod-becker WDYT?

Reply:
@dannyzaken I did my testing in openshift 4.13.0 and nodeTaintsPolicy was considered for topology-spread-constraints, need to get a confirmation. 
@liranmauda without nodeTaintsPolicy master node also consider for skew calculation, since our skew value is maxSkew is 1 it might give unexpected behavior. It will not affect the pod deployment because of `whenUnsatisfiable: ScheduleAnyway`

Reply:
Seems like openshift 4.13 is based on Kubernetes 1.12 [see this link](https://access.redhat.com/solutions/4870701)
So I guess we are good.

Reply:
I could find the `nodeTaintsPolicy` in the Deployment creation page in openshift 4.13, So `nodeTaintsPolicy`  should be available.
![image](https://github.com/noobaa/noobaa-operator/assets/79908956/c1d6660b-c543-4a3f-a857-06fde2d61c94)

Reply:
Did testing with or without `nodeTaintsPolicy : Honor` in openshift 4.13(2 master nodes, 4 worker nodes), and could find the following behavior
- without `nodeTaintsPolicy : Honor`, the master node is also considered for `Skew` calculation, and if `whenUnsatisfiable: DoNotSchedule`(just for testing purposes) we can deploy only 4 pods in 4 worker nodes. This is because if we try to deploy one more pod it will violate topologySpreadConstraints that include the master node.
- with `nodeTaintsPolicy : Honor`, the master node is not considered for `Skew` calculation, and if `whenUnsatisfiable: DoNotSchedule`(just for testing purposes) we can deploy as any pod in 4 worker nodes and the only constrain is CPU/Memory.
with this behavior, I am certain that `nodeTaintsPolicy : Honor` is valid for openshift 4.13 and the latest releases.

Reply:
@naveenpaul1 Just wanted to say that I also stumbled this before, but I had a lower openshift version so couldn't use the `nodeTaintsPolicy: Honor` option, so my solution was to label the worker nodes with a label `noobaa-endpoint-domain: <node-name>` (uniq values) and used that label for the topologyKey. 

I also saw on 4.10 and 4.12 that the deployment yaml doesn't even store this new option. So what will happen for deployments that upgrade from 4.12 or lower? I will never be set... Perhaps we should reconcile this to the deployment if missing (but not override it if exists). WDYT @dannyzaken

Reply:
@guymguym Yes it makes sense for clusters migrating from 4.12 and older versions of openshift because there wont be `nodeTaintsPolicy` in the endpoint deployment yaml file when these clusters migrate.

Reply:
@naveenpaul1 But our operator can be installed on clusters with lower version. We are not syncing the version of the noobaa operator with the cluster version. So if we deploy the new operator on a 4.12 version, that setting will get lost.

Reply:
The thing is that upstream does not have the same notion of OCP -1 for k8s.
We could reconcile it, but then I worry that the operator will keep creating the topologySpreadConstraints in case a user is interested in removing it or modifying it...
@dannyzaken what would you do?

Reply:
We should provide a flag to disable it under endpointSpec. I usually prefer to avoid adding configurations in our CR, but I don't see a way around it for older k8s versions. @guymguym @naveenpaul1 WDYT?

Reply:
I think if we are planning to give users an option to edit/delete topology constraints we do not have another option, especially when we add the code in noobaa endpoint reconcile loop.

Reply:
@naveenpaul1, following our discussion, let's add an env variable to disable this feature. It's better than adding a flag to the CRD